conmon-rs/server/src/pause.rs

```diff
@@ -0,0 +1,265 @@
+        debug!("Initializing pause");
+
+        let mut args = vec![];
+        let mut namespaces = vec![];
+        for namespace in init_namespaces.iter() {
+            match namespace? {
+                conmon::Namespace::Ipc => {
+                    args.push("--ipc");
+                    namespaces.push(Namespace::Ipc);
+                }
+                conmon::Namespace::Net => {
+                    args.push("--net");
+                    namespaces.push(Namespace::Net);
+                }
+                conmon::Namespace::Pid => {
+                    args.push("--pid");
+                    namespaces.push(Namespace::Pid);
+                }
+                conmon::Namespace::Uts => {
+                    args.push("--uts");
+                    namespaces.push(Namespace::Uts);
+                }
+                conmon::Namespace::User => {
+                    warn!("Unsharing the user namespace is not supported yet");
+                    // args.push("--user");
+                    // namespaces.push(Namespace::User);
+                }
+            }
+        }
+        debug!("Pause namespaces: {:?}", namespaces);
+
+        let path = PathBuf::from(PAUSE_PATH).join(Uuid::new_v4().to_string());
+        fs::create_dir_all(&path).context("create base path")?;
+        debug!("Pause base path: {}", path.display());
+
+        let program = env::args().next().context("no args set")?;
+        let mut child = Command::new(program)
+            .arg("pause")
+            .arg("--path")
+            .arg(&path)
+            .args(args)
+            .spawn()
+            .context("run pause")?;
+
+        let status = child.wait().context("wait for pause child")?;
+        if !status.success() {
+            bail!("exit status not ok: {status}")
+        }
+
+        let pid = fs::read_to_string(path.join(PAUSE_PID_FILE))
+            .context("read pause PID path")?
+            .trim()
+            .parse::<u32>()
+            .context("parse pause PID")?;
+        info!("Pause PID is: {pid}");
+
+        Ok(Self {
+            path,
+            namespaces,
+            pid: Pid::from_raw(pid as pid_t),
+        })
+    }
+
+    /// Run a new pause instance.
+    pub fn run<T: AsRef<Path> + Copy>(
+        path: T,
+        ipc: bool,
+        pid: bool,
+        net: bool,
+        user: bool,
+        uts: bool,
+    ) -> Result<()> {
+        let mut namespaces = vec![];
+        let mut flags = CloneFlags::empty();
+        if ipc {
+            flags.insert(CloneFlags::CLONE_NEWIPC);
+            namespaces.push(Namespace::Ipc);
+        }
+        if pid {
+            flags.insert(CloneFlags::CLONE_NEWPID);
+            namespaces.push(Namespace::Pid);
+        }
+        if net {
+            flags.insert(CloneFlags::CLONE_NEWNET);
+            namespaces.push(Namespace::Net);
+        }
+        if user {
+            // flags.insert(CloneFlags::CLONE_NEWUSER);
```

Comment:
left over? if not, I think we could use a comment

Reply:
Yep, added a comment.

Reply:
We can add support for userns at the next iteration before releasing 0.5.0. Should be straightforward to implement.